jenkins/bootstrap.py

```diff
@@ -735,11 +737,16 @@ def job_args(args):
     return [os.path.expandvars(a) for a in args]
 
 
-def job_script(job, use_json):
+def job_script(job, use_json, jobs_dir):
     """Return path to script for job."""
+    jobs_dir_resolver = lambda f: test_infra('jobs/%s' % f)
+
+    if jobs_dir:
+        jobs_dir_resolver = lambda f: os.path.join(jobs_dir, f)
+
     if not use_json:
```

Comment:
There are two ways bootstrap.py operates:

* deprecated way: --job=FOO looks for a FOO.sh file somewhere. No new jobs should use this pattern, and we will soon delete this code path.
* supported way: --job=FOO looks for the FOO job in the config.json file somewhere. It then does what the scenario specifies (which could just be executing a script).